### PR TITLE
stm32l4-multi: Add aes driver (as libaes) + fix RTC alarm

### PR DESCRIPTION
**Pull request by Darchiv**

### CODE REVIEW

File: multi/stm32l4-multi/lib/libaes.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+	for (i = 0; i < 4; i++) {
+		*addr = byte_reverse(*((unsigned int *) vector));
+		vector += 4;
+		addr -= 1;
+	}
+}
+
+
+void libaes_set_key(const unsigned char *key, int keylen)
```
Comment (agkaminski) [inline multi/stm32l4-multi/lib/libaes.c:62]: Use camelCase (libaes_setKey)

File: multi/stm32l4-multi/lib/libaes.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+{
+	if (keylen == aes_128) {
+		store_vector4(keyr0, key);
+		*(common.base + cr) &= ~(1 << 18);
+	} else {
```
Comment (agkaminski) [inline multi/stm32l4-multi/lib/libaes.c:67]: else should be in the new line

File: multi/stm32l4-multi/lib/libaes.c [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		store_vector4(keyr4, key);
+		store_vector4(keyr0, key + 16);
+		*(common.base + cr) |= (1 << 18);
+	}
+
+	common.is_key_scheduled = 0;
+}
+
+
+void libaes_set_iv(const unsigned char *iv)
```
Comment (agkaminski) [inline multi/stm32l4-multi/lib/libaes.c:77]: Use camelCase

File: multi/stm32l4-multi/lib/libaes.c [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+{
+	store_vector4(ivr0, iv);
+}
+
+
+static void derive_decryption_key(void)
```
Comment (agkaminski) [inline multi/stm32l4-multi/lib/libaes.c:83]: Use camelCase

File: multi/stm32l4-multi/lib/libaes.c [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+		out += 4;
+	}
+
+	*(common.base + cr) |= (1 << 7);
+}
+
+
+int libaes_init(void)
+{
+	common.base = (void*) 0x50060000;
```
Comment (agkaminski) [inline multi/stm32l4-multi/lib/libaes.c:145]: (void *)0x50060000;

**Review by agkaminski — COMMENTED** [review]
Apart from coding conventions LGTM. RTC fix should be really in separate PR, but it's ok

File: multi/stm32l4-multi/lib/libaes.c (hunk 1 above)
Comment (Darchiv) [reply multi/stm32l4-multi/lib/libaes.c]: Done

File: multi/stm32l4-multi/lib/libaes.c (hunk 2 above)
Comment (Darchiv) [reply multi/stm32l4-multi/lib/libaes.c]: Done

File: multi/stm32l4-multi/lib/libaes.c (hunk 3 above)
Comment (Darchiv) [reply multi/stm32l4-multi/lib/libaes.c]: Done

File: multi/stm32l4-multi/lib/libaes.c (hunk 4 above)
Comment (Darchiv) [reply multi/stm32l4-multi/lib/libaes.c]: Done

File: multi/stm32l4-multi/lib/libaes.c (hunk 5 above)
Comment (Darchiv) [reply multi/stm32l4-multi/lib/libaes.c]: Done

**Pull request merged**